### DISCUSSION TITLE
[braintree-web] BraintreeError doesn't extend Error

### DISCUSSION THREAD

**Discussion by someguynamedmatt** [DefinitelyTyped/DefinitelyTyped#66396] [Issues with a @types package]
Should `BraintreeError` ([def found here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/braintree-web/core.d.ts)) extend the `Error` interface?

Right now, it doesn't which becomes tricky when you're attempting to pass a definite error into anything like a logging library which typically expects an error-like object.

Proposed:
```typescript
export type BraintreeErrorTypes = 'CUSTOMER' | 'MERCHANT' | 'NETWORK' | 'INTERNAL' | 'UNKNOWN';

interface BraintreeError extends Error {
        /**
     * @description A code that corresponds to specific errors.
     */
    code: string;

    /**
     * @description A short description of the error.
     */
    message: string;

    /**
     * @description The type of error.
     */
    type: BraintreeErrorTypes;

    /**
     * @description Additional information about the error, such as an underlying network error response.
     */
    details: any;
}
```

[a comment by an automated account is not shown]

**Comment by daelmaak** [comment]
Hey Matt, that ofc depends on whether `braintree-web` itself extends `Error`. I found that [it does](https://github.com/braintree/braintree-web/blob/831be9d4f7bea387cae98f22c8f215854c994b75/src/lib/braintree-error.js#L52) so imo your request is in essence valid.

**Reply by someguynamedmatt** [reply]
@daelmaak Should I submit a PR?

**Reply by daelmaak** [reply]
Yes please @someguynamedmatt, PR is always appreciated.
